### PR TITLE
Add initial Literate documentation

### PR DESCRIPTION
**Pull request by JeffFessler**
I am looking for an incomplete Cholesky decomposition precoditioner and I found this package.
To help others (and myself) figure out how to use this code some docs would be helpful.
Here is a PR with at least some basics.

### CODE REVIEW

File: docs/lit/examples/1-overview.jl
Comment (mohdibntarek) [inline docs/lit/examples/1-overview.jl:66]: seems odd to have that in docs

**Comment by mohdibntarek** [comment]
I don't know Literate enough to review this. But it seems CI isn't giving a preview of the docs.

File: docs/lit/examples/1-overview.jl
Comment (JeffFessler) [reply docs/lit/examples/1-overview.jl:66]: Agreed.  I had meant to post an issue to avoid this in the long run.
Now I have: #33.

**Comment by JeffFessler** [comment]
> But it seems CI isn't giving a preview of the docs.

I forgot to include the CI for the docs.  It is added now.
There are settings to force the CI to make a preview of the docs but I don't usually use that because it seems to make the repo get larger.  I check the preview locally by running `julia --project=. make.jl` in the `docs/` directory.  But to make it easier for you, for now I changed the one line in `make.jl` to be `push_preview = true` so now it should build a doc preview.

The docs are not too exciting now because this package has so few docstrings...

**Comment by JeffFessler** [comment]
BTW, the docs will deploy only if someone who is a maintainer enables github pages with appropriate ssh deploy keys under Settings tab.  See https://documenter.juliadocs.org/stable/man/hosting/#GitHub-Actions

**Comment by JeffFessler** [comment]
I tried to manually create a gh-pages branch at the command line and push that. Then I reran the CI.
The CI ran, but did not create any preview in the gh-pages branch.
I think it can work only if the maintainers set that up under Settings, and then rerun the workflow.

**Comment by mohdibntarek** [comment]
I setup the keys. Let's see.

**Comment by mohdibntarek** [comment]
You can now preview the docs at https://julialinearalgebra.github.io/Preconditioners.jl/previews/PR31

**Comment by mohdibntarek** [comment]
Now it also shows up in the checks under `documenter/deploy` which you can click to preview the docs.

File: docs/lit/examples/1-overview.jl
Comment (JeffFessler) [reply docs/lit/examples/1-overview.jl:66]: btw, if you want to wait until #33 is resolved to merge this, it is ok with me.
i ended up using `LLDL` directly for now...  this package helped me discover LLDL so thanks for that!

Comment (mohdibntarek) [reply docs/lit/examples/1-overview.jl:66]: I think allocating the result matrix once and doing inplace ldiv! calls with views will be more efficient. As I mentioned in the issue, it might be useful to upstream these implementations. But I think having any sort of docs here is probably more important than supporting an additional feature.

Comment (JeffFessler) [reply docs/lit/examples/1-overview.jl:66]: Good point.  For now I just updated the comments in the example to alert the reader to issue #33.
We can fix this up later if/when that issue is addressed.

**Comment by JeffFessler** [comment]
I see that it is running all the CI tests, even when only docs are changed.  Eventually the CI file should be updated to prevent that...

**Review by mohdibntarek — APPROVED** [review]
Will merge and let's iterate on it.

**Comment by JeffFessler** [comment]
I forgot to update the README with links to the docs.  It will be a while before I have time to do that though.

**Pull request merged**
